rhino/src/main/java/org/mozilla/javascript/NativeDate.java

```diff
@@ -988,24 +988,51 @@ private static double parseISOString(Context cx, String s) {
                 // milli secs are different, digit 2 and 3 are optional
                 int value = 0;
                 int digitsFound = 0;
+                boolean shouldRound = false;
+
                 for (; i < len; i++) {
                     char c = s.charAt(i);
                     if (c < '0' || c > '9') {
                         break;
                     }
 
-                    // skip more digits
                     if (digitsFound < 3) {
                         value = 10 * value + (c - '0');
                         digitsFound++;
+                    } else if (digitsFound == 3) {
+                        // Use 4th digit for rounding decision
+                        shouldRound = (c >= '5');
+                        digitsFound++;
+                        // Skip remaining digits
                     }
                 }
+
                 if (digitsFound == 0) {
                     state = ERROR;
                     break loop;
                 }
+
                 if (digitsFound < 3) {
                     value = value * (digitsFound == 1 ? 100 : 10);
+                } else if (shouldRound) {
+                    value++;
+                    // Handle overflow: 1000ms -> add 1 second
```

Comment:
Thanks for this contribution - maybe you can add some test cases for the various code paths regarding overflow handling